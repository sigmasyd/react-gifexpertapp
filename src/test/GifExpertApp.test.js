import React from 'react';
const { shallow } = require('enzyme');
const { GifExpertApp } = require('../GifExpertApp');

describe('Se realizan pruebas sobre el componente principal GifExpertApp', () => {
  test('debe generarse correctamente el componente <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de mostrar una lista de categorias', () => {
    // actualmente no hay forma de evaluar el useState, por lo que
    // se recurre a una categoria por defecto lo cual es valido para las pruebas
    // -------->
    // si tengo dos elementos en la categoria por default, deberia
    // de tener dos elementos GifGrid, es decir:
    // tantos elementos GifGrid's como categorias tenga
    const categories = ['Saint Seiya', 'Transformer'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
