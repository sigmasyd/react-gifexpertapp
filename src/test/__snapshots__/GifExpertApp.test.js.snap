// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se realizan pruebas sobre el componente principal GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Saint Seiya"
      key="Saint Seiya"
    />
    <GifGrid
      category="Transformer"
      key="Transformer"
    />
  </ol>
</Fragment>
`;

exports[`Se realizan pruebas sobre el componente principal GifExpertApp debe generarse correctamente el componente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
